### PR TITLE
Markdown mode cleanup and content sync

### LINKED ISSUE #203: Page scrolls while typing large texts in markdown mode
Ref: https://github.com/bigbinary/neeto-kb-web/issues/1121

@AbhayVAshokan _a
high-priority _t

### PR DESCRIPTION
**Pull request by labeebklatif**

### CODE REVIEW

**Comment by labeebklatif** [comment]
@AbhayVAshokan Please have a look whether all required changes/fixes in the https://github.com/bigbinary/neeto-editor-tiptap/pull/204 is present in this PR so that once this gets merged, we can close https://github.com/bigbinary/neeto-editor-tiptap/pull/204

File: lib/components/Editor/index.jsx [hunk 1]
```diff
@@ -93,17 +99,34 @@ const Editor = (
...
+  const markdownEditor = useMarkdownEditor({
+    content: initialValue,
+    onUpdate: (markdown) => onChange(markdownToHtml(markdown)),
+    onSubmit: (markdown) => onSubmit && onSubmit(markdownToHtml(markdown)),
+  });
+
   /* Make editor object available to the parent */
   React.useImperativeHandle(ref, () => ({
-    editor: editor,
+    editor: markdownMode ? markdownEditor : editor,
```
[the remark this answers is not in the document]
Comment (labeebklatif) [reply lib/components/Editor/index.jsx:116]: I also favour the first approach

**Comment by AbhayVAshokan** [comment]
@labeebklatif can you please test this PR by replacing App.jsx with this [gist](https://gist.github.com/AbhayVAshokan/850dc525b3fa5801de13a2c3744f7532) too? I believe this is close to the implementation in KB.

**Comment by labeebklatif** [comment]
currently, the character count is hidden in markdown mode and the `textarea` is being used as an uncontrolled component to fix issue with height

@karthiknmenon _a Please review

cc: @AbhayVAshokan

**Comment by labeebklatif** [comment]
PR #204 can be closed once this is successfully merged

**Review by karthiknmenon — APPROVED** [review]

**Pull request merged**